### ISSUE TITLE
Stop using deprecated libkiwix methods

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-desktop#805]
libkiwix `10.x.y` have deprecated many methods and function which are still in used in Kiwix Desktop. This has two consequences:
* Many warning at Kiwix Desktop compilation time
* Stop us to removed these old/deprecated primitives in future libkiwix releases.

Would be beneficial to remove the usage of these deprecated methods and use modern/new API of libkiwix.

The effort has already been made in Kiwix Tools, see https://github.com/kiwix/kiwix-tools/pull/535 and this PR might help to know what to do with Kiwix Desktop.

**Comment by juuz0** [comment]
@kelson42 Yes, been working on it. I should be able to do this and other PR by Saturday (college exam on Friday)
